sdk/Lusid.Sdk.Tests/Utilities/ApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Net;
+using System.Threading;
+using Lusid.Sdk.Api;
+using Lusid.Sdk.Client;
+using Lusid.Sdk.Utilities;
+using NUnit.Framework;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Tests.Utilities
+{
+    [TestFixture]
+    public class ApiRetryHandlerTests
+    {
+        private LusidApiFactory _apiFactory;
+        private readonly Policy<IRestResponse> _initialRetryPolicy = RetryConfiguration.RetryPolicy;
+        private HttpListener _httpListener;
+        private const string ListenerUriPrefix = "http://localhost:4444/";
+        
+        [SetUp]
+        public void SetUp()
+        {
+            _httpListener = new HttpListener();
+            _httpListener.Prefixes.Add(ListenerUriPrefix);
+            
+            var testApiConfig = TestLusidApiFactoryBuilder.CreateApiConfiguration(null);
+            testApiConfig.ApiUrl = ListenerUriPrefix;
+            
+            _apiFactory = new LusidApiFactory(testApiConfig);
+        }
+
+        [Test]
+        [Explicit("This passes locally, but not in the pipeline, and no idea why. Setting to explicit until resolved.")]
+        public void CallGetPortfoliosApi_WhenPollyRetryPolicyConfigured_PollyIsTriggered()
+        {
+            const int expectedNumberOfRetries = ApiRetryHandler.MaxRetryAttempts;
+            _httpListener.Start();
+            for (var i = 0; i < expectedNumberOfRetries + 1; i++)
+            {
+                 _httpListener.BeginGetContext(result =>
+                {
+                    var listener = (HttpListener) result.AsyncState;
+                    // Call EndGetContext to complete the asynchronous operation.
+                    var context = listener.EndGetContext(result);
+
+                    // Obtain a response object.
+                    var response = context.Response;
+
+                    // Abort the response. This returns 0 status code.
```

Comment:
When would this happen in reality? A `0` status code being returned?

Reply:
When connection is aborted, or the thing timeouts. Also there are many case scenarios we don't need the root cause of, that's why we want to see the stack trace with these retries, at least temporarily so we are familiar with the failures

Reply:
Ok